napari/utils/_tests/test_translations.py

```diff
@@ -0,0 +1,98 @@
+"""Test the translations API."""
+
+import subprocess
+import sys
+from pathlib import Path
+
+from ..translations import _get_display_name, _is_valid_locale, translator
+
+TEST_LOCALE = "es_CO"
+HERE = Path(__file__).parent
+TEST_LANGUAGE_PACK_PATH = HERE / "napari-language-pack-es_CO"
+
+
+def test_locale_valid():
+    # Install test language package
+    p = subprocess.Popen(
```

Comment:
Ended up installing the package as part of the test @tlambert03 🤷🏼

Reply:
this seems a little heavy. i'm not sure i have a better idea if the other installation methods didn't work. i wonder if there are any potential negative consequences if the installation fails, will the whole test run fail, or everything else will keep going as expected. I am ok with this though if we don't have other options. We can also just go with this for now and then maybe revisit sometime in the future, so no need to hold up the PR in my opinion

Reply:
yeah, lemme look into this before we merge.  will be busy for the weekend though.

Reply:
Yeah, now that I see how this is being used... I don't love it.  We've even run into issues (https://github.com/napari/napari/issues/1009) in the past by including complete "subpackages" with their own `setup.py` as test fixtures.  Tests also shouldn't have side-effects, and this test will permanently modify the user's development environment... which is less than desirable.

We have to do a similar sort of entry point [discovery test](https://github.com/napari/napari-plugin-engine/blob/master/testing/test_discovery.py) in napari-plugin-engine.  There I use the pytest `tmp_path` fixture to build a temporary plugin fixture that gets cleaned up at the end.  I can't remember off the top of my head how I get that discovered during tests, but it's possible and far preferable to literally running pip install in a subprocess).  If you're tired of messing around making these tests work, let me know and I can make a PR to this PR.

Also mentioned this elsewhere, but we can probably reuse a lot of the napari-plugin-engine mechanisms for translation plugin discovery here.

Reply:
Thanks for the input,  the fixture can have a teardown and uninstall de test, so that would solve the side effect issue.

I do not have much experience with tox to be honest, but I did try different ideas to no avail.

>Also mentioned this elsewhere, but we can probably reuse a lot of the napari-plugin-engine mechanisms for translation plugin discovery here.

More on that on those comments.